option.go

```diff
@@ -163,6 +163,28 @@ func KeepFields[T any](fields ...string) Option {
 	})
 }
 
+// KeepExported transforms a value of struct type T. It makes a copy of its input,
+// preserving exported fields only.
+//
+// This effectively makes comparison use only exported fields.
+//
+// See also Transform.
+func KeepExported[T any]() Option {
+	t := reflect.TypeOf((*T)(nil)).Elem()
+	var fields []string
+	for _, f := range reflect.VisibleFields(t) {
```

Comment:
I think VisibleFields is not what you want here. Because it will return fields of embedded types as well. We only need to do the fields of this type itself.

So this loop should be something like `for i := 0; i < t.NumField(); i++`.